### ISSUE TITLE
java.lang.IllegalStateException: Unable to create a system terminal

### ISSUE THREAD

**Issue by holgerbrandl** [ctongfei/progressbar#31]
When running the following Kotlin program
```
object Foo {
    @JvmStatic
    fun main(args: Array<String>) {
        LogManager.getLogManager().getLogger("").level = Level.FINEST

        ProgressBar.wrap(((1..10000).toList()), "Preparing Images").forEach { sleep(10) }
    }
}
```
in Jetbrains Intellj IDEA (v2018.1), I get the following error:
```
java.lang.IllegalStateException: Unable to create a system terminal
	at org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:276)
	at org.jline.terminal.TerminalBuilder.build(TerminalBuilder.java:222)
	at org.jline.terminal.TerminalBuilder.terminal(TerminalBuilder.java:62)
	at me.tongfei.progressbar.ProgressThread.<init>(ProgressThread.java:37)
	at me.tongfei.progressbar.ProgressBar.<init>(ProgressBar.java:51)
	at me.tongfei.progressbar.ProgressBar.<init>(ProgressBar.java:29)
	at me.tongfei.progressbar.wrapped.ProgressBarWrappedIterator.<init>(ProgressBarWrappedIterator.java:18)
	at me.tongfei.progressbar.wrapped.ProgressBarWrappedIterable.iterator(ProgressBarWrappedIterable.java:24)
	at Foo.main(ConvNet.kt:149)
	Suppressed: java.lang.NoClassDefFoundError: com/sun/jna/Platform
		at org.jline.terminal.impl.jna.JnaNativePty.current(JnaNativePty.java:40)
		at org.jline.terminal.impl.jna.JnaSupportImpl.current(JnaSupportImpl.java:16)
		at org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:320)
		... 8 more
	Caused by: java.lang.ClassNotFoundException: com.sun.jna.Platform
		at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
		at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
		at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:335)
		at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
		... 11 more
	Suppressed: java.lang.UnsupportedOperationException
		at org.jline.terminal.impl.jansi.JansiSupportImpl.current(JansiSupportImpl.java:71)
		at org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:330)
		... 8 more
	Suppressed: java.io.IOException: Not a tty
		at org.jline.terminal.impl.ExecPty.current(ExecPty.java:44)
		at org.jline.terminal.TerminalBuilder.doBuild(TerminalBuilder.java:339)
		... 8 more
	Caused by: java.io.IOException: Error executing 'tty': not a tty
		at org.jline.utils.ExecHelper.exec(ExecHelper.java:42)
		at org.jline.terminal.impl.ExecPty.current(ExecPty.java:41)
		... 9 more

Preparing Images  10% │██▎                  │  1093/10000 (0:00:12 / 0:01:38) 
```

The same does not report a stacktrace when being run as a java program in IDEA.

**Comment by ctongfei** [comment]
This is probably a bug in the terminal emulator in IDEA under Kotlin mode.

**Comment by holgerbrandl** [comment]
To report it to JetBrains I'd need to know of what nature the bug is. Should the terminal emulator report certain properties, implement some interface, ...?

**Comment by ctongfei** [comment]
Try calling this:
```java
org.jline.terminal.TerminalBuilder.terminal()
```

see https://github.com/ctongfei/progressbar/blob/master/src/main/java/me/tongfei/progressbar/ProgressThread.java#L37

**Comment by ctongfei** [comment]
Also in 0.7 the `printStackTrace()` is suppressed so you would not see the stacktrace anymore.

**Comment by Nickersoft** [comment]
Currently getting this error on zsh on Mac as well... any updates? Tried calling `org.jline.terminal.TerminalBuilder.terminal()` but to no avail.

**Comment by ctongfei** [comment]
@Nickersoft Are you also getting a `java.lang.IllegalStateException`?

**Comment by Nickersoft** [comment]
Yep, I am

**Comment by ctongfei** [comment]
@Nickersoft Apart from showing the exception message, is the progress bar working fine? If it is, I'm just going to suppress the output

**Comment by ctongfei** [comment]
Which version worked?

**Comment by Nickersoft** [comment]
Last version that worked I believe is 0.5.5

**Comment by ctongfei** [comment]
Got it. This should be because the updated dependency of `jline`.

**Comment by ctongfei** [comment]
Got it. This should be because of the updated dependency of `jline`.

**Comment by Nickersoft** [comment]
What should? The issue is being caused by the latest version of `jline`?

**Comment by ctongfei** [comment]
Yes. `progressbar` utilizes the `TerminalFactory` in `jline`, which is changed from `jline` 2.x to 3.x. In 3.x more checks are performed. I've updated `jline` from 2.x to 3.x in `progressbar` 0.6.0 so that's why 0.5.5 is the last version that works for you.

So I'd like to confirm: if you are not running it in a test suite (just normally), besides printing out the `java.lang.IllegalStateException` message, is the progress bar running?

**Comment by ctongfei** [comment]
This should be fixed along with #40. Closed for now.
